### PR TITLE
flags: allow to mix --long and --tree

### PR DESCRIPTION
**Pull request by Monkeypac**
This is a feature that I'd love to have in `lsd`. 

I just recently discovered `lsd` ~a few weeks ago.
I'm not a Rust developer, this is the first time I do some so feel free to criticize, I just want to suggest the solution.

Refs: #185
Closes: #204

### CODE REVIEW

File: src/core.rs [hunk 1]
```diff
@@ -64,8 +64,13 @@ impl Core {
 
     fn fetch(&self, paths: Vec<PathBuf>) -> Vec<Meta> {
         let mut meta_list = Vec::with_capacity(paths.len());
+        let tree_layout_true = Layout::Tree { long: true };
+        let tree_layout_false = Layout::Tree { long: false };
 
-        let depth = if self.flags.recursive || self.flags.layout == Layout::Tree {
+        let depth = if self.flags.recursive
```
Comment (meain) [inline src/core.rs]: I guess we could switch it out to this.

```rust
        let depth = match &self.flags.layout {
            &Layout::Tree { long } if long || !long || self.flags.recursive => {
                self.flags.recursion_depth
            }
            _ => 1,
        };
```

Comment (Monkeypac) [reply src/core.rs]: Thx, I was looking for a better solution but couldn't get it ahah.

**Comment by meain** [comment]
Hmm, looks like clippy is complaining about that code. This should be good.

```rust
        let depth = match self.flags.layout {
            Layout::Tree { .. } => self.flags.recursion_depth,
            _ if self.flags.recursive => self.flags.recursion_depth,
            _ => 1,
        };
```

**Comment by Monkeypac** [comment]
oh yeah I didn't see your answer, thanks ahah

**Pull request merged**